biz.aQute.bndlib/src/aQute/bnd/cdi/CDIAnnotations.java

```diff
@@ -62,11 +69,56 @@ public boolean analyzeJar(Analyzer analyzer) throws Exception {
 		Instructions instructions = new Instructions(header);
 		Collection<Clazz> list = analyzer.getClassspace()
 			.values();
+		Packages contained = analyzer.getContained();
+
+		Parameters bcp = analyzer.getBundleClassPath();
+		Jar currentJar = analyzer.getJar();
+		List<String> beanArchives = new ArrayList<>();
+		beanArchives.add(getName(currentJar.getName(), currentJar.getVersion()));
+
+		for (Entry<String, Attrs> entry : bcp.entrySet()) {
+			String path = entry.getKey();
+			Resource resource = currentJar.getResource(path);
+			if (resource != null) {
+				Jar jar = Jar.fromResource(path, resource);
+
+				Resource beansResource = jar.getResource("META-INF/beans.xml");
+
+				Discover discover = findDiscoveryMode(beansResource);
+
+				if (discover != Discover.none) {
+					beanArchives.add(getName(path, jar.getVersion()));
+				}
+			}
+		}
+
 		List<String> names = new ArrayList<>();
 		TreeSet<String> provides = new TreeSet<>();
 		TreeSet<String> requires = new TreeSet<>();
 
 		for (Clazz c : list) {
+			if (c.isEnum() || c.isInterface() || c.isAnnotation() || c.isInnerClass() || c.isSynthetic()
```

Comment:
Since you exclude interfaces, you don't need to also exclude annotations since an annotation is an interface.